app/src/organisms/ProtocolSetupParameters/ChooseNumber.tsx

```diff
@@ -0,0 +1,166 @@
+    String(parameter.value)
+  )
+
+  // We need to arbitrarily set the value of the keyboard to a string the
+  // same length as the initial parameter value (as string) when the component mounts
+  // so that the delete button operates properly on the exisiting input field value.
+  const [prevKeyboardValue, setPrevKeyboardValue] = React.useState<string>('')
+  React.useEffect(() => {
+    const arbitraryInput = new Array(paramValue).join('*')
+    // @ts-expect-error keyboard should expose for `setInput` method
+    keyboardRef.current?.setInput(arbitraryInput)
+    setPrevKeyboardValue(arbitraryInput)
+  }, [])
+
+  if (parameter.type !== 'int' && parameter.type !== 'float') {
+    console.log(`Incorrect parameter type: ${parameter.type}`)
+    return null
+  }
+  const handleClickGoBack = (newValue: number): void => {
+    if (error != null) {
+      makeSnackbar(t('value_out_of_range_generic'))
+    } else {
+      setParameter(newValue, parameter.variableName)
+      handleGoBack()
+    }
+  }
+
+  const handleKeyboardInput = (e: string): void => {
+    if (prevKeyboardValue.length < e.length) {
+      const lastDigit = e.slice(-1)
+      if (
+        !'.-'.includes(lastDigit) ||
+        (lastDigit === '.' && !paramValue.includes('.')) ||
+        (lastDigit === '-' && paramValue.length === 0)
+      ) {
+        setParamValue(paramValue + lastDigit)
+      }
+    } else {
+      setParamValue(paramValue.slice(0, paramValue.length - 1))
+    }
+    setPrevKeyboardValue(e)
+  }
+
+  const paramValueAsNumber = Number(paramValue)
+  const resetValueDisabled = parameter.default === paramValueAsNumber
+  const { min, max } = parameter
+  const error =
+    paramValue === '' ||
+    Number.isNaN(paramValueAsNumber) ||
+    paramValueAsNumber < min ||
+    paramValueAsNumber > max
+      ? t(`value_out_of_range`, {
+          min: parameter.type === 'int' ? min : min.toFixed(1),
+          max: parameter.type === 'int' ? max : max.toFixed(1),
+        })
+      : null
+
+  return (
+    <>
+      <ChildNavigation
+        header={i18n.format(parameter.displayName, 'sentenceCase')}
+        onClickBack={() => {
+          handleClickGoBack(paramValueAsNumber)
+        }}
+        buttonType="tertiaryLowLight"
+        buttonText={t('restore_default')}
+        onClickButton={() =>
+          resetValueDisabled
+            ? makeSnackbar(t('no_custom_values'))
+            : setParamValue(String(parameter.default))
+        }
+      />
+      <Flex
+        alignSelf={ALIGN_CENTER}
+        gridGap={SPACING.spacing48}
+        paddingX={SPACING.spacing40}
+        paddingBottom={SPACING.spacing40}
+        marginTop="7.75rem"
+        height="22rem"
+        justifyContent={JUSTIFY_CENTER}
+        alignItems={ALIGN_CENTER}
+      >
+        <Flex
+          width="30.5rem"
+          height="100%"
+          gridGap={SPACING.spacing24}
+          flexDirection={DIRECTION_COLUMN}
+          marginTop="7.75rem"
```

Comment:
probably using justify-content is better.

```suggestion
         justifyContent={JUSTIFY_CENTER}
```

Reply:
The only thing here is that if an error message appears, the content moves. What do we think about that?

Reply:
oh I see.
that makes sense.
could you add note here about why we need to keep marginTop?